src/ai/backend/manager/repositories/vfolder/repository.py

```diff
@@ -94,6 +103,64 @@ async def get_by_id(self, vfolder_id: uuid.UUID) -> Optional[VFolderData]:
                 return None
             return self._vfolder_row_to_data(vfolder_row)
 
+    @repository_decorator()
+    async def get_allowed_vfolder_hosts(
+        self, user_uuid: uuid.UUID, group_uuid: Optional[uuid.UUID]
+    ) -> str:
+        """
+        Get the allowed VFolder hosts for a user.
+        """
+        async with self._db.begin_readonly_session() as db_session:
+            if group_uuid:
+                group_row: Optional[GroupRow] = await db_session.scalar(
+                    sa.select(GroupRow).where(GroupRow.id == group_uuid)
+                )
+                if group_row is None:
+                    raise GroupNotFound(object_name="Group", object_id=group_uuid)
+
+                return group_row.allowed_vfolder_hosts
```

Comment:
It seems better to separate the function that receives user_uuid and the function that only receives group_uuid.